gallifreyan.js

```diff
@@ -88,6 +89,24 @@ function updateText() {
     generateWords(words);
 }
 
+//discard any unfinished manual actions (line addition/deletion)
+function resetModes() {
+    deleteLineMode  = false;
+    if (addLineMode) {
+        addLineMode = false;
+        deleteLine(selectedLine);
+        selectedLine = null;
+    }
+}
+
+//create a new line and let user to position it
+function addNewLine() {
+    addLineMode = true;
+    selectedLine = new Line(allCircles[0], -PI/2, allCircles[0], -PI/2);
+    allCircles[0].lines.removeItem(selectedLine);
```

Comment:
You left it in the new commit, do you want it to stay after all? (I can merge now)